src/pages/DisplayFixtures.jsx

```diff
@@ -1,66 +1,55 @@
-import { Fragment, useState, useMemo, useReducer } from 'react'
+import { Fragment, useState, useMemo } from 'react'
 import { useNavigate } from 'react-router-dom';
 import '../css/Scores.css'
 import getDates from '../functions/getDates';
 import { dateFormatter2 } from '../functions/dateTimeFormatter'
 import useApp from '../hooks/useApp'
-import predictionsReducer from '../common/PredictionsReducer';
 import FixtureList from '../components/FixtureList';
-import { UserTotalPoints } from '../components/UserTotalPoints'
+import { UserTotalPoints } from '../components/UserTotalPoints';
+import { getPredictions } from '../functions/getPredictions.jsx';
 
 // This component sets up the fixtures and determines which elements are shown
 // At this point, the user has selected the competition and round
 // The next step is to display the relevant fixtures, the predictions (if any) and either
 // - match status (if complete or in play)
 // - date/time (if non started)
 const DisplayFixtures = () => {
-  const { allPredictions, setAllPredictions, fixtures, selectedCompetition, results, round, currentUser, userPredictions } = useApp();
+  const { allPredictions, setAllPredictions, fixtures, selectedCompetition, round, currentUser, userPredictions, setUserPredictions } = useApp();
   const [editMode, setEditMode] = useState(false);
+  const [isSaving, setIsSaving] = useState(false);
   const [matchesStarted, setMatchesStarted] = useState(false);
   const [totalPoints, setTotalPoints] = useState(0);
+  const [changedPredictions, setChangedPredictions] = useState([]);
   const comp = selectedCompetition;
   const nav = useNavigate();
 
-
   // initially filter all predictions and return those belonging to the user
-  const deleteFunc = useMemo(() => {
+  useMemo(() => {
+    if (!(Array.isArray(allPredictions) && currentUser)) return [];
+
     return allPredictions.filter(prediction => {
-      return prediction.user && prediction.user._id === currentUser._id;
+      return prediction.user?._id === currentUser._id;
     });
-  }, [allPredictions, currentUser._id]);
-
-
-  const [predictions, dispatch] = useReducer(predictionsReducer, userPredictions);
-
+  }, [allPredictions, currentUser]);
 
   function handleUpdatePrediction(prediction) {
-    dispatch({
-      type: 'updatedPrediction',
-      prediction: prediction
-    })
+    const newUserPredictions = userPredictions.map((p) => {
+      if (p.fixture_id === prediction.fixture_id) {
+        // console.log(`Updated prediction: ` + JSON.stringify(prediction))
+        const unrelatedChangedPredictions = changedPredictions.filter(pred => pred._id !== prediction._id);
+        setChangedPredictions([...unrelatedChangedPredictions, prediction]);
+        return prediction;
+      } else {
+        return p;
+      }
+    });
+    setUserPredictions(newUserPredictions);
```

Comment:
insteed of using dispatch and updating `predictions` state variable, we can simply update `userPredictions` via `setUserPredictions` so we maintain only one source of truth!